src/main/java/frc/robot/subsystems/drive/SwerveModule.java

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems.drive;
+
+import com.revrobotics.AbsoluteEncoder;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxAbsoluteEncoder.Type;
+import com.revrobotics.SparkMaxPIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.util.sendable.Sendable;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj.Timer;
+import frc.robot.Calibrations;
+import frc.robot.Constants;
+import frc.robot.utils.AbsoluteEncoderChecker;
+import frc.robot.utils.SparkMaxUtils;
+
+public class SwerveModule implements Sendable {
+  public final CANSparkMax drivingSparkMax;
+  public final CANSparkMax turningSparkMax;
+
+  private final RelativeEncoder drivingEncoder;
+  private final AbsoluteEncoder turningEncoder;
+  private AbsoluteEncoderChecker turningAbsoluteEncoderChecker = new AbsoluteEncoderChecker();
+
+  private final SparkMaxPIDController drivingPIDController;
+  private final SparkMaxPIDController turningPIDController;
+
+  private double chassisAngularOffsetRadians = 0;
+  private SwerveModuleState desiredState = new SwerveModuleState(0.0, new Rotation2d());
+
+  /**
+   * Constructs a SwerveModule and configures the driving and turning motor, encoder, and PID
+   * controller. This configuration is specific to the REV MAXSwerve Module built with NEOs, SPARKS
+   * MAX, and a Through Bore Encoder.
+   */
+  public SwerveModule(int drivingCanId, int turningCanId, double chassisAngularOffset) {
+    drivingSparkMax = new CANSparkMax(drivingCanId, MotorType.kBrushless);
+    turningSparkMax = new CANSparkMax(turningCanId, MotorType.kBrushless);
+    chassisAngularOffsetRadians = chassisAngularOffset;
+
+    drivingEncoder = drivingSparkMax.getEncoder();
+    drivingPIDController = drivingSparkMax.getPIDController();
+    turningEncoder = turningSparkMax.getAbsoluteEncoder(Type.kDutyCycle);
+    turningPIDController = turningSparkMax.getPIDController();
+
+    desiredState.angle = new Rotation2d(turningEncoder.getPosition());
+    drivingEncoder.setPosition(0);
+  }
+
+  /** Does all the initialization for the spark */
+  void initTurnSpark() {
+
+    turningSparkMax.restoreFactoryDefaults();
+    turningSparkMax.setInverted(false);
```

Comment:
ok so TURNING_ENCODER_INVERTED is a variable in Constants.java and is set to true. it is used elsewhere so you can't use that. I would make a new variable for this value of false